frontend/src/components/SeedlotRegistrationSteps/OrchardStep/index.tsx

```diff
@@ -0,0 +1,570 @@
+  const invalidOrchardValue = 'Please insert a valid orchard id between 0 and 999';
+
+  const refControl = useRef<any>({});
+
+  const [additionalOrchard, setAdditionalOrchard] = useState<boolean>(false);
+
+  const [responseBody, setResponseBody] = useState<SeedlotOrchard>(orchardData);
+
+  const [invalidOrchardId, setInvalidOrchardId] = useState<boolean>(false);
+  const [invalidOrchardMessage, setInvalidOrchardMessage] = useState<string>('');
+  const [invalidAddOrchardId, setInvalidAddOrchardId] = useState<boolean>(false);
+  const [invalidAddOrchardMessage, setInvalidAddOrchardMessage] = useState<string>('');
+  const [invalidFemGametic, setInvalidFemGametic] = useState<boolean>(false);
+  const [invalidMalGametic, setInvalidMalGametic] = useState<boolean>(false);
+  const [invalidBreeding, setInvalidBreeding] = useState<boolean>(false);
+
+  const addRefs = (element: HTMLInputElement, name: string) => {
+    if (element !== null) {
+      refControl.current = {
+        ...refControl.current,
+        [name]: element
+      };
+    }
+  };
+
+  const setResponse = (field: string[], value: string[] | boolean) => {
+    const isStrArray = Array.isArray(value);
+
+    if (field.length === 1) {
+      setResponseBody({
+        ...responseBody,
+        [field[0]]: isStrArray ? value[0] : value
+      });
+    } else {
+      // It only get here if we are setting 2 fields,
+      // so there is no need for a 'for' or a 'map'
+      setResponseBody({
+        ...responseBody,
+        [field[0]]: isStrArray ? value[0] : value,
+        [field[1]]: isStrArray ? value[1] : value
+      });
+    }
+  };
+
+  const validateOrchardId = (event: React.ChangeEvent<HTMLInputElement>, nameField: string) => {
+    const { value, name } = event.target;
+    if (value) {
+      axios.get(getUrl(ApiAddresses.OrchardRetriveOne).replace(':orchardnumber', value), getAxiosConfig())
+        .then((response) => {
+          if (response.data.orchard) {
+            // Clear any errors, if any
+            if (name === 'orchardId' && invalidOrchardId) {
+              setInvalidOrchardId(false);
+              setInvalidOrchardMessage(invalidOrchardValue);
+            } else if (invalidAddOrchardId) {
+              setInvalidAddOrchardId(false);
+              setInvalidAddOrchardMessage(invalidOrchardValue);
+            }
+
+            setResponse([name, nameField], [value, response.data.orchard.name]);
+
+          // Set error messages for id not found
+          } else if (name === 'orchardId') {
+            setInvalidOrchardMessage(orchardIdNotFound);
+            setInvalidOrchardId(true);
+          } else {
+            setInvalidAddOrchardMessage(orchardIdNotFound);
+            setInvalidAddOrchardId(true);
+          }
+        })
+        .catch((error) => {
+          // eslint-disable-next-line
+          console.error(`Error: ${error}`);
+        });
+    } else if (name === 'orchardId') {
+      setInvalidAddOrchardId(true);
```

Comment:
Credit to sonarcloud, this `if` block does the same thing as the `else` under it, we should combine them